solidity/script/DeployAVS.s.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+import "forge-std/Script.sol";
+
+import {IDelegationManager} from "../contracts/interfaces/avs/IDelegationManager.sol";
+import {ECDSAStakeRegistry} from "../contracts/avs/ECDSAStakeRegistry.sol";
+import {Quorum} from "../contracts/interfaces/avs/IECDSAStakeRegistryEventsAndErrors.sol";
+import {HyperlaneServiceManager} from "../contracts/avs/HyperlaneServiceManager.sol";
+
+contract DeployAVS is Script {
+    address public avsDirectory;
+    IDelegationManager public delegationManager;
+    address public paymentCoordinator;
+
+    Quorum quorum;
+    uint256 thresholdWeight = 6667;
+
+    function loadEigenlayerAddresses(string targetEnv) external {
+        string memory root = vm.projectRoot();
+        string memory path = string.concat(
+            root,
+            "solidity/script/avs/eigenlayer_addresses.json"
+        );
+        string json = vm.readFile(path);
+
+        avsDirectory = json.readAddress(
+            string(abi.encodePacked(".", targetEnv, ".avsDirectory"))
+        );
+        delegationManager = IDelegationManager(
+            json.readAddress(
+                string(abi.encodePacked(".", targetEnv, ".delegationManager"))
+            )
+        );
+        paymentCoordinator = json.readAddress(
+            string(abi.encodePacked(".", targetEnv, ".paymentCoordinator"))
+        );
```

Comment:
why not `vm.parseJson` and abi.decode

Reply:
readAddress does parseJson behind the scenes, it's a nice helper. I'm using parseJson for reading the StrategyInfo.